frontend/src/Components/modals/VerfiyResidentModal.tsx

```diff
@@ -0,0 +1,133 @@
+import { forwardRef, useState } from 'react';
+import { CRUDModalProps, FormInputTypes, FormModal, Input } from '.';
+import { FieldValues, SubmitHandler } from 'react-hook-form';
+import {
+    Facility,
+    ResidentEngagementProfile,
+    ServerResponseOne,
+    ValidResident
+} from '@/common';
+import API from '@/api/api';
+import { useCheckResponse } from '@/Hooks/useCheckResponse';
+import { useLoaderData } from 'react-router-dom';
+
+export const VerifyResidentModal = forwardRef(function (
+    {
+        mutate,
+        onSuccess,
+        target
+    }: CRUDModalProps<ResidentEngagementProfile> & {
+        onSuccess: (resident: ValidResident) => void;
+    },
+    verifyResidentModal: React.ForwardedRef<HTMLDialogElement>
+) {
+    const [errors, setErrors] = useState('');
+    const [facility, setFacility] = useState('');
+    const facilities = useLoaderData() as Facility[];
+    const checkResponse = useCheckResponse({
+        mutate: mutate,
+        refModal: verifyResidentModal
+    });
+    const executeResidentCheck: SubmitHandler<FieldValues> = async (data) => {
+        if (!validateFacility()) {
+            return;
+        }
+        const response = (await API.get(
+            `users/resident-verify?user_id=${target?.user.id}&doc_id=${data.doc_id}&facility_id=${facility}`
+        )) as ServerResponseOne<ValidResident>;
+        checkResponse(
+            response.success,
+            'Resident not found, try again',
+            'Resident found, please confirm transfer'
+        );
```

Comment:
Should the error message say something about the ID not matching the resident? The resident was found but the validation check failed.